docs/migration-guide.md

```diff
@@ -2,36 +2,59 @@
 
-The group ID of the Readium modules is now `org.readium.kotlin-toolkit`, for instance:
+The modules now target Android SDK 34. If your app also targets it, you will need the `FOREGROUND_SERVICE_MEDIA_PLAYBACK` permission in your `AndroidManifest.xml` file to use TTS and audiobook playback.
 
-```groovy
-dependencies {
-    implementation "org.readium.kotlin-toolkit:readium-shared:$readium_version"
-    implementation "org.readium.kotlin-toolkit:readium-streamer:$readium_version"
-    implementation "org.readium.kotlin-toolkit:readium-navigator:$readium_version"
-    implementation "org.readium.kotlin-toolkit:readium-opds:$readium_version"
-    implementation "org.readium.kotlin-toolkit:readium-lcp:$readium_version"
-}
+### `Publication`
+
+#### Opening a `Publication`
+
+The `Streamer` object has been deprecated in favor of components with smaller responsibilities:
+
+* `AssetRetriever` grants access to the content of an asset located at a given URL, such as a publication package, manifest, or LCP license
+* `PublicationOpener` uses a publication parser and a set of content protections to create a `Publication` object from an `Asset`.
+
+[See the user guide for a detailed explanation on how to use these new APIs](guides/open-publication.md).
+
+#### Sharing `Publication` across Android activities
+
+The `putPublication` and `getPublication` helpers in `Intent` are deprecated. Now, it is the application's responsibility to pass `Publication` objects between activities and reopen them when necessary.
+
+You can take a look at the [`ReaderRepository` in the Test App](https://github.com/readium/kotlin-toolkit/blob/09e338b0f3acc8d59282280bded6c4bf93de6281/test-app/src/main/java/org/readium/r2/testapp/reader/ReaderRepository.kt#L46) for inspiration.
+
+Alternatively, you can copy [the deprecated helpers](https://github.com/readium/kotlin-toolkit/blob/7649378a0a6b924abedf8b72372c3808fe9b992f/readium/shared/src/main/java/org/readium/r2/shared/extensions/Intent.kt#L26) and add them to your codebase. However, please note that this approach is discouraged because it will not handle configuration changes smoothly.
+
+### `MediaType`
+
+#### Sniffing a `MediaType`
+
+`MediaType` no longer has static helpers for sniffing it from a file or URL. Instead, you can use an `AssetRetriever` to retrieve the format of a file.
+
+```kotlin
+val httpClient = DefaultHttpClient()
+val assetRetriever = AssetRetriever(context.contentResolver, httpClient)
+
+val mediaType = assetRetriever.sniffFormat(File(...))
+    .getOrElse { /* Failed to access the asset or recognize its format */ }
+    .mediaType
 ```
 
-### Migration of HREFs and Locators (bookmarks, annotations, etc.)
+### HREFs
+
+#### `Link.href` and `Locator.href` are not strings anymore
+
+`Link.href` and `Locator.href` are now respectively `Href` and `Url` objects. If you still need the string value, you can call `toString()`, but you may find the `Url` objects more useful in practice.
```

Comment:
How do we get an `Url` from `Link.href`?

Reply:
With `link.url()`, which is an alias of `link.href.resolve()`.

Reply:
Haha, _I_ know. I think the reader might have the same question though.